Sources/FoundationEssentials/String/String+Internals.swift

```diff
@@ -19,6 +19,63 @@ internal import Foundation_Private.NSString
 import Darwin
 #endif
 
+#if os(Windows)
+import WinSDK
+
+extension String {
+    package func withNTPathRepresentation<Result>(_ body: (UnsafePointer<WCHAR>) throws -> Result) throws -> Result {
+        guard !isEmpty else {
+            throw CocoaError.errorWithFilePath(.fileReadInvalidFileName, "")
+        }
+
+        // 1. Normalize the path first.
+
+        var path = self
+
+        // Strip the leading `/` on a RFC8089 path (`/[drive-letter]:/...` ).  A
+        // leading slash indicates a rooted path on the drive for the current
+        // working directory.
+        var iter = path.makeIterator()
+        if iter.next() == "/", iter.next()?.isLetter ?? false, iter.next() == ":" {
+            path.removeFirst()
+        }
+
+        // Win32 APIs can support `/` for the arc separator. However,
+        // symlinks created with `/` do not resolve properly, so normalize
+        // the path.
+        path.replace("/", with: "\\")
+
+        // Drop trailing slashes unless it follows a drive specification.  The
+        // trailing arc separator after a drive specifier indicates the root as
+        // opposed to a drive relative path.
+        while path.count > 1, path.last == "\\" {
+            if path.count == 3, path[0].isLetter, path[1] == ":" {
```

Comment:
`path` here is ` String` (right?) in which case I wouldn't expect `Int` subscripting to work - does this compile on Windows for some reason that I'm missing, or do we need to calculate these two indices (might be nice to do that calculation on a separate line above to break it up a bit)

Reply:
Hmm, I wonder if I missed the errors ... there are still ~150 errors during compilation so its somewhat difficult to always catch everything. I can use `path.first?.isLetter`, and just compute the second index I suppose.